checks/datadog.py

```diff
@@ -175,8 +175,7 @@ def check(self, agentConfig, move_end=True):
             self._events = []
 
             # Build our tail -f
-            if self._gen is None:
-                self._gen = TailFile(self.logger, self.log_path, self._line_parser).tail(line_by_line=False, move_end=move_end)
+            self._gen = TailFile(self.logger, self.log_path, self._line_parser).tail(line_by_line=False, move_end=move_end)
```

Comment:
I don't think we need to create a TailFile object every iteration, probably just make sure we call `tail()`.... Can we look into doing that properly? The way I see it maybe that should just be in the constructor.... Let me know if you see problems with that approach.